pepper-apis/src/test/java/org/broadinstitute/ddp/route/GetUserAnnouncementsRouteTest.java

```diff
@@ -177,4 +189,74 @@ public void test_basicContentStyle() {
+                .then().assertThat()
+                .statusCode(200).contentType(ContentType.JSON)
+                .body("$.size()", equalTo(1))
+                .body("[0].guid", not(isEmptyOrNullString()))
+                .body("[0].permanent", equalTo(true))
+                .body("[0].message", equalTo(msgTemplate.getTemplateText()))
+                .extract().path("[0].guid");
+
+        TransactionWrapper.useTxn(handle -> {
+            List<UserAnnouncement> found = handle.attach(UserAnnouncementDao.class)
+                    .findAllForUserAndStudy(testData.getUserId(), testData.getStudyId())
+                    .collect(Collectors.toList());
+            assertEquals(1, found.size());
+            assertEquals(guid, found.get(0).getGuid());
+        });
+
+        given().auth().oauth2(token)
+                .pathParam("userGuid", testData.getUserGuid())
+                .pathParam("studyGuid", testData.getStudyGuid())
+                .when().get(url)
+                .then().assertThat()
+                .statusCode(200).contentType(ContentType.JSON)
+                .body("$.size()", equalTo(1))
+                .body("[0].guid", equalTo(guid));
+    }
+
+    @Test
+    public void test_formerProxy_stillGetMessages() {
+        StudyDto study = TransactionWrapper.withTxn(handle -> {
+            StudyDto testStudy = TestDataSetupUtil.generateTestStudy(handle, RouteTestUtil.getConfig());
+            handle.attach(JdbiClientUmbrellaStudy.class).insert(testData.getClientId(), testStudy.getId());
+
+            UserGovernanceDao userGovernanceDao = handle.attach(UserGovernanceDao.class);
+            Governance gov = userGovernanceDao.createGovernedUserWithGuidAlias(testData.getClientId(), testData.getUserId());
+            userGovernanceDao.grantGovernedStudy(gov.getId(), testStudy.getId());
+            handle.attach(JdbiUserStudyEnrollment.class)
+                    .changeUserStudyEnrollmentStatus(gov.getGovernedUserId(), testStudy.getId(), EnrollmentStatusType.REGISTERED);
+
+            handle.attach(UserAnnouncementDao.class).insert(gov.getProxyUserId(), testStudy.getId(), msgTemplate.getTemplateId(), true);
+            userGovernanceDao.disableProxy(gov.getId());
+
+            return testStudy;
+        });
+
+        given().auth().oauth2(token)
+                .pathParam("userGuid", testData.getUserGuid())
+                .pathParam("studyGuid", study.getGuid())
+                .when().get(url)
+                .then().assertThat()
+                .statusCode(200).contentType(ContentType.JSON)
+                .body("$.size()", equalTo(1))
+                .body("[0].guid", not(isEmptyOrNullString()))
+                .body("[0].permanent", equalTo(true))
+                .body("[0].message", equalTo(msgTemplate.getTemplateText()));
+
+        TransactionWrapper.useTxn(handle -> {
+            handle.attach(UserAnnouncementDao.class).deleteAllForUserAndStudy(testData.getUserId(), study.getId());
```

Comment:
I think `deleteAllForUserAndStudy()` is covered by the `@After`?

Reply:
In this test case, we created a new study which is different from the one `@After` looks at, so we're calling delete for this one.

Reply:
Got it!